app/services/users.js

```diff
@@ -2,15 +2,48 @@ const db = require('../models/index');
 
 const logger = require('../logger');
 const { databaseError, badRequest_Error } = require('../errors');
+const { roleAdmin } = require('../constants');
 
-const { signUp, dataComplete, getUsers } = require('../serializers/users');
+const { pagination } = require('../serializers/users');
 const { encryptar, comparePassword } = require('../helpers/utils');
 
 exports.saveUser = async data => {
   try {
     data.password = encryptar(data.password);
     const newUser = await db.User.create(data);
-    return signUp(newUser);
+    return newUser;
+  } catch (err) {
+    logger.error(databaseError(err.errors));
+    throw databaseError(err.errors);
+  }
+};
+
+exports.saveUserAdmin = async data => {
```

Comment:
I think that this logic should only be in charge of updating the user's role if it already exists, the other could be validated in the controller using the methods that already exist in the service